### PR TITLE
Fix documentation of qiskit.circuit.classicalfunction

### LINKED ISSUE #9334: ClassicalFunction compiler (qiskit.circuit.classicalfunction) Overview Sentence trails off without completion
### Where is the problem?
https://qiskit.org/documentation/apidoc/classicalfunction.html
### What is the problem?
Sentence in Overview trails off without completion:
> Following Qiskit’s little-endian bit ordering convention, the left-most bit (a) is the most significant bit and the right-most bit (d) is the least significant bit. The resulting

... and that's it.
### What would you expect?
Finish the thought!

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes #9334

It's hard to finish the thought from 28 months ago (#4522), but I think the sentence was trying to explain the in endianness. So I extended the example to make that point.

### CODE REVIEW

File: qiskit/circuit/classicalfunction/__init__.py [hunk 1]
```diff
@@ -34,10 +34,22 @@
...
+           d: ──o──
+                │  
+           c: ──■──
+                │  
+           b: ──o──
+                │  
+           a: ──■──
+              ┌─┴─┐
+      return: ┤ X ├
+              └───┘
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply qiskit/circuit/classicalfunction/__init__.py:49]: This, indeed, looks like a bug in `.synth(registerless=False)`:

```python
quantum_circuit = grover_oracle.synth(registerless=False)
print(quantum_circuit.draw())
```
```
     c: ──■──
          │  
     b: ──■──
          │  
     a: ──o──
        ┌─┴─┐
return: ┤ X ├
        └───┘
```

**Comment by 1ucian0** [comment]
It seems to me that we accidentally trip into a bug here #9363 . So, `on hold` until a fix for it. Thanks @Cryoris !

**Comment by 1ucian0** [comment]
blocked by https://github.com/Qiskit/qiskit-terra/pull/9536

**Comment by 1ucian0** [comment]
https://github.com/Qiskit/qiskit/pull/9536 is merged. removing `on hold` . The example now is valid @Cryoris

**Review by jakelishman — APPROVED** [review]
This has been stalled for a while, but looks fine to me.

**Pull request merged**